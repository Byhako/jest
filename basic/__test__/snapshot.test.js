import { shot } from '../snapshot';
import data from '../toto.json';

describe('Probando snapshot', () => {
  test('Verifica a Toto', () => {
    expect(shot(data)).toMatchSnapshot();
  });

  test('Siempre fallo :(', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random()*20)
    };
    expect(user).toMatchSnapshot();
  });

  test('Excepcion', () => {
    const user = {
      createAt: new Date(),
      id: 15
    };
    expect(user).toMatchSnapshot({
      createAt: expect.any(Date)
    })
  });
});

