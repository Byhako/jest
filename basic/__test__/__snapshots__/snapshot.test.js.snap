// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando snapshot Excepsion 1`] = `
Object {
  "createAt": Any<Date>,
  "id": 15,
}
`;

exports[`Probando snapshot Siempre fallo :( 1`] = `
Object {
  "createAt": 2020-08-16T01:24:11.956Z,
  "id": 2,
}
`;

exports[`Probando snapshot Verifica a Toto 1`] = `
Object {
  "gender": "Macho",
  "id": "1",
  "name": "Toto",
  "species": "Cat",
  "status": "Alive",
}
`;
